qiskit_aer/quantum_info/states/aer_state.py

```diff
@@ -0,0 +1,306 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+State class that handles internal C++ state safely
+"""
+import numpy as np
+from qiskit.providers.aer.backends.controller_wrappers import AerStateWrapper
+from ...backends.aerbackend import AerError
+
+
+class AerState:
+    """Internal class to access state of Aer."""
+
+    def __init__(self):
+        """State that handles cpp quantum state safely"""
+        self._shift_to_created_state()
+        self._state = AerStateWrapper()
+        self._method = 'statevector'
+        self._last_qubit = -1
+        self._initialized = False
+        self._closed = False
```

Comment:
Do you think it would be useful to have options to set the method and stuff at construction time instead of needing to call `configure()`